### ISSUE TITLE
Add non-deny-all tests to e2e_test.go

### ISSUE THREAD

**Issue by willbeason** [open-policy-agent/frameworks#154]
Prerequisite: https://github.com/open-policy-agent/frameworks/pull/150

See `constraint/pkg/client/e2e_test.go`.

The current tests in e2e_test.go all require the CT under test to reject all objects being reviewed. Instead, these tests should include cases specifying CTs that don't reject objects, or conditionally reject them. (Since these are E2E/integration tests and not unit tests, these paths should be covered)

This likely requires moving around how expectations are set - moving `got` from being specified inline in the test to `denyAllCases` (which needs to be renamed).

**Comment by willbeason** [comment]
Duplicate of #154
